### PR TITLE
CLN: Added static types

### PR DESCRIPTION
**Pull request by ShaharNaveh**
- [ ] closes #xxxx
- [ ] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

---

I'll revert any change related to styling, I applied "black" formatting in some places, just so I could think straight.

### CODE REVIEW

File: pandas/_libs/tslibs/parsing.pyx [hunk 1]
```diff
@@ -467,8 +473,14 @@ cdef inline object _parse_dateabbr_string(object date_string, object default,
...
 
 
-cdef dateutil_parse(str timestr, object default, ignoretz=False,
-                    tzinfos=None, dayfirst=None, yearfirst=None):
+cdef dateutil_parse(
+    str timestr,
+    object default,
+    bint ignoretz=False,
+    tzinfos=None,
+    dayfirst=None,
```
Comment (jreback) [inline pandas/_libs/tslibs/parsing.pyx]: bint no?

File: pandas/_libs/tslibs/parsing.pyx [hunk 2]
```diff
@@ -467,8 +473,14 @@ cdef inline object _parse_dateabbr_string(object date_string, object default,
     raise ValueError(f'Unable to parse {date_string}')
 
 
-cdef dateutil_parse(str timestr, object default, ignoretz=False,
-                    tzinfos=None, dayfirst=None, yearfirst=None):
+cdef dateutil_parse(
+    str timestr,
+    object default,
+    bint ignoretz=False,
+    tzinfos=None,
```
Comment (jreback) [inline pandas/_libs/tslibs/parsing.pyx]: str/object

File: pandas/_libs/tslibs/parsing.pyx [hunk 3]
```diff
@@ -760,15 +780,20 @@ def _format_is_iso(f) -> bint:
     return False
 
 
-def _guess_datetime_format(dt_str, dayfirst=False, dt_str_parse=du_parse,
-                           dt_str_split=_DATEUTIL_LEXER_SPLIT):
+def _guess_datetime_format(
+    dt_str,
```
Comment (jbrockmendel) [inline pandas/_libs/tslibs/parsing.pyx]: str?

File: pandas/_libs/tslibs/parsing.pyx (hunk 1 above)
Comment (MomIsBestFriend) [reply pandas/_libs/tslibs/parsing.pyx]: Doing so will require changing the default value of ```dayfirst```, is it okay?

File: pandas/_libs/tslibs/parsing.pyx (hunk 3 above)
Comment (MomIsBestFriend) [reply pandas/_libs/tslibs/parsing.pyx]: Python, raises on tests with this change.

**Review by WillAyd — APPROVED** [review]
lgtm

**Comment by jbrockmendel** [comment]
lgtm

File: pandas/_libs/tslibs/parsing.pyx (hunk 1 above)
Comment (jbrockmendel) [reply pandas/_libs/tslibs/parsing.pyx:481]: yes please.  ping on green

**Comment by WillAyd** [comment]
@jreback any outstanding comments?

**Comment by jreback** [comment]
thanks @MomIsBestFriend

**Pull request merged**